### DISCUSSION TITLE
VLESS+gRPC+Reality

### DISCUSSION THREAD

**Discussion by MolecularDust** [XTLS/Xray-core#4127] [Q&A]
Many guides consider RPRX-Vision as a sort of gold standard for using with Reality. I've tested other transports with Reality and my personal experience is that HTTP2 and especially gRPC yield slightly lower latency (10-15% in my tests) and make a heavy use of YouTube, for example, overall snappier, there are less waiting spinner animations. But do HTTP2/gRPC pose a higher risk for detection? Specifically, are they prone to TLS-in-TLS issue when used with Reality?

**Comment by donnyxray** [comment]
Yes to all of that. It has lower latency, is also more easily detected, and therefore discouraged.

That said, we offer our users multiple protocols, using a dedicated IP pool for each protocol. The percentage of blocked IP's is very similar, currently around 4% in each pool.

Sometimes it drops to 2% or jumps up to more than 20%. But it is always similar across the different protocol pools. Factors other than the protocol used seem to be more significant.
